docs/DeveloperGuide.md

```diff
@@ -1,12 +1,220 @@
+
+It implements the following operations:
+
+- setUnfavourite() - Removes a carpark from the favourite list.
+
+Given below is an example of how the Favourite class is used to perform favourite / unfavourite operations:
+
+Step 1. On startup, the Favourite is initialised with a directory and file path to the file that contains data, and a 
+favouriteList attribute.
+The Favourite class calls `updateFavouriteList()` shortly after to populate its ArrayList with the carpark IDs in the 
+file.
+
+Step 2. The user executes `favourite 1` to favourite the carpark with carpark ID `1`. The command is passed to the 
+Parser class which returns a Command with value `FAVOURITE`.
+The main program first checks if the second argument in the command is equal to `list`. If so, the main program calls 
+`showList()` and passes it output to the Ui class for printing.
+Else, the main program checks if there exists a carpark with carpark ID `1` by calling `findCarpark()` from the 
+CarparkList class, which throws `NoCarparkFoundException` if no carpark was found.
+`setFavourite()` is then called to add the valid carpark ID into `favouriteList`, which searches `favouriteList` for 
+any identical carpark IDs and throws `DuplicateCarparkException` if found, to prevent addition of duplicates.
+`1` is then added into `favouriteList` and `setFavourite()` calls `writeFavouriteList()` which overwrites data from 
+`favouriteList` to the favourites file.
+
+> Note: If any exception is thrown, `1` will not be added into `favouriteList`, and `writeFavouriteList()` will not be 
+> called, hence preserving the validity of the carpark IDs.
+
+Step 3. The user executes `favourite 2` to favourite the carpark with carpark ID `2`. Same as Step 2.
+
+Step 4. The user realises he/she made a mistake and wants to unfavourite the carpark with carpark ID `2`, and executes 
+`unfavourite 2`.
+The command is passed to the Parser class which returns a Command with value `UNFAVOURITE`. The main program calls 
+`setUnfavourite()`,
+which first checks if `favouriteList` contains an entry that matches `2`, and throws `NoCarparkFoundException` if none 
+is found.
+Next, `2` is removed from `favouriteList` and `setUnfavourite()` calls `writeFavouriteList()` which overwrites data from
+`favouriteList` to the favourites file.
+
+Step 5. The user wants to view all favourited carparks and executes `favourite list`. The command is passed to the 
+Parser class which returns a Command with value `FAVOURITE`.
+The main program calls `showList()` this time as the second argument in the command is `list`. The `showList()` method
+uses StringBuilder to format all the contents of `favouriteList` into a user-friendly string, and returns it for 
+printing.
+
+The following sequence diagrams shows how a favourite / unfavourite command works:
+
+![Favourite Sequence Diagram](images/FavouriteSequenceDiagram.png)
```

Comment:
Maybe you can make the diagram bigger for easier reading. The diagram is also not sharp.